### PR TITLE
Move ASB to new project system

### PR DESCRIPTION
**Pull request by SimonCropp**
Connects to https://github.com/Particular/NServiceBus.Azure/issues/302

### CODE REVIEW

**Comment by SimonCropp** [comment]
@Particular/azure-maintainers need some help on this one. just updated to the new version of the core. were those tests sources previously manipulated in any way?

**Comment by SeanFeldman** [comment]
@SimonCropp probably would be easier to upgrade first to the new csproj file format and then update to the newer core.

There's a pending PR for `develop` (PR #570). Will this PR cause merge conflicts?

**Comment by SeanFeldman** [comment]
@SimonCropp you've also changed the version of NUnit (from [3.6.1](https://github.com/Particular/NServiceBus.AzureServiceBus/pull/573/files#diff-8221554ae9be37c4864acc99c8e055e1L58) to [3.7.1](https://github.com/Particular/NServiceBus.AzureServiceBus/pull/573/files#diff-8221554ae9be37c4864acc99c8e055e1R20)). Running these tests locally throws as well at very unexpected place, where it shouldn't:

image

Exception on comparison of two objects that have values

image

I suggest to upgrade the project format first, then the dependencies.

File: src/NServiceBus.AzureServiceBus.sln.DotSettings [hunk 1]
```diff
@@ -597,6 +597,7 @@ II.2.12 &lt;HandlesEvent /&gt;&#xD;
 	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002EJavaScript_002ECodeStyle_002ESettingsUpgrade_002EJsCodeFormatterSettingsUpgrader/@EntryIndexedValue">True</s:Boolean>
 	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002EJavaScript_002ECodeStyle_002ESettingsUpgrade_002EJsParsFormattingSettingsUpgrader/@EntryIndexedValue">True</s:Boolean>
 	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002EJavaScript_002ECodeStyle_002ESettingsUpgrade_002EJsWrapperSettingsUpgrader/@EntryIndexedValue">True</s:Boolean>
+	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002EXml_002ECodeStyle_002EFormatSettingsUpgrade_002EXmlMoveToCommonFormatterSettingsUpgrade/@EntryIndexedValue">True</s:Boolean>
```
Comment (SeanFeldman) [inline src/NServiceBus.AzureServiceBus.sln.DotSettings]: Should enable C# 7, just like [for ASQ](https://github.com/Particular/NServiceBus.Persistence.AzureStorage/pull/150/files#diff-c35417364bb6af4a4ba0b9d6b7625052R3)

Comment (SimonCropp) [inline src/NServiceBus.AzureServiceBus.sln.DotSettings]: ok. but that is not within scope of this PR

**Comment by SimonCropp** [comment]
> There's a pending PR for develop (PR #570). Will this PR cause merge conflicts?

probably. but i suggest u merge this first since it will be easier to re-apply #570 than it will be to re do this.

**Comment by SimonCropp** [comment]
@SeanFeldman reverted core and nunit for test projects and

**Comment by SeanFeldman** [comment]
Looks much better. The only failing test (deferral test) is a flaky one which I'll disable anyways.
What's more interesting is the NUnit odd behavior.

**Comment by SeanFeldman** [comment]
Thank you @SimonCropp

**Comment by SimonCropp** [comment]
Thanks sean

**Pull request merged**